external-crates/move/crates/move-compiler/src/typing/core.rs

```diff
@@ -1698,6 +1728,19 @@ fn report_visibility_error(
             diag.add_secondary_label((test_loc, test_msg))
         }
     }
+    if let Some(names) = context.expanding_macros_names() {
+        let macro_s = if context.macro_expansion.len() > 1 {
+            "macros"
+        } else {
+            "macro"
+        };
+        diag.add_note(format!(
+            "This visibility error occurs in a macro body while by expanding the {macro_s} {names}"
```

Comment:
A little confused on "while" here. Do I need to (in my head) sum the lines together?

Reply:
Whoops, "while by" was a typo -- I updated it to say say "macro body while expanding". Does that make more sense?

Reply:
Yup, much better.